src/panels/KaitoPanel.ts

```diff
@@ -117,45 +139,57 @@ export class KaitoPanelDataProvider implements PanelDataProvider<"kaito"> {
         });
     }
     private async handleKaitoInstallation(webview: MessageSink<ToWebViewMsgDef>) {
-        // Check if Kaito pods already exist
-        if (this.filterKaitoPodNames.length > 0) {
-            vscode.window.showInformationMessage(
-                `Kaito pods already exist in the cluster ${this.clusterName}. Skipping installation.`,
-            );
-            return;
-        }
-        // Register feature
-        const subscriptionFeatureRegistrationType = {
-            properties: {},
-        };
-        const options = {
-            subscriptionFeatureRegistrationType,
-        };
-
-        const featureRegistrationPoller = await longRunning(`Registering the AIToolchainOperator.`, () => {
-            return this.featureClient.subscriptionFeatureRegistrations.createOrUpdate(
-                "Microsoft.ContainerService",
-                "AIToolchainOperatorPreview",
-                options,
-            );
-        });
+        // Get the feature registration state
+        const getFeatureClientRegisterState = await longRunning(
+            `Getting the AIToolchainOperator registration state.`,
+            () => {
+                return this.featureClient.features.get("Microsoft.ContainerService", "AIToolchainOperatorPreview");
+            },
+        );
 
-        if (featureRegistrationPoller.properties?.state !== "Registered") {
-            webview.postKaitoInstallProgressUpdate({
-                operationDescription: "Installing Kaito",
-                event: 3,
-                errorMessage: "Failed to register feature",
-                models: [],
+        if (getFeatureClientRegisterState.properties?.state !== "Registered") {
```

Comment:
💡 Self note: all this need some refactor which we can do in the final drop - basically , bunch of Private functs and `longRunning` job for better loop back.